internal/engine/wazevo/backend/isa/arm64/instr.go

```diff
@@ -814,7 +832,7 @@ func (i *instruction) String() (str string) {
 	case cCmpImm:
 		panic("TODO")
 	case fpuMov64:
-		str = fmt.Sprintf("mov %s.8b, %s.8b", formatVRegSized(i.rd.nr(), 128), formatVRegSized(i.rn.nr(), 128))
+		str = fmt.Sprintf("mov %s.8b, %s.8b", formatVRegSized(i.rd.nr(), 64), formatVRegSized(i.rn.nr(), 64))
```

Comment:
this changes fpuMov64 to 128bit vregs for printing. 

the issue is that popcnt MOVs a 64-bit GPR to a vector register.

Was assuming 128 bits here incorrect in the first place or do we need to add stronger logic to autodetect? 

- if it was incorrect, then I will fix the test cases that I broke; 
- otherwise, how to do this without breaking the String() routine? Can we detect the size of the register "automatically" instead of asserting it here?

The offending line is the panic in `formatVRegSized()`:

```
		switch r.RegType() {
		case regalloc.RegTypeInt:
			switch size {
			case 32:
				ret = fmt.Sprintf("w%d?", r.ID())
			case 64:
				ret = fmt.Sprintf("x%d?", r.ID())
			default:
				panic("BUG: invalid register size: " + strconv.Itoa(int(size))) /// <--- fault occurs here
			}

```

Reply:
yeah you are right, this should be `vX.b8` instead of `qX.b8`. I will fix it in a separate PR shortly so that we can keep this small

Reply:
128 bit is correct.  See how `fpuMov64` is encoded: 


- https://shell-storm.org/online/Online-Assembler-and-Disassembler/?opcodes=f21fbf0e&arch=arm64&endianness=little&baddr=0x00000000&dis_with_addr=True&dis_with_raw=True&dis_with_ins=True#disassembly
- https://github.com/tetratelabs/wazero/blob/main/internal/engine/wazevo/backend/isa/arm64/instr_encoding_test.go#L219

If you need to move GP to Vector, then use unimplemented `movToVec`. `fpuMov64` is not for them.

Reply:
so in other words, you do not need to modify `fpuMov64`, and implement `movToVec` or `movFromVec`